### PR TITLE
[WIP] Pull to refresh

### PR DESCRIPTION
**Pull request by ChristophWurst**
https://github.com/apeatling/web-pull-to-refresh/

### CODE REVIEW

**Comment by jancborchardt** [comment]
Nice! :)

In the [demo](https://apeatling.com/demos/web-ptr/) of that library you can only pull via actually clicking and pulling. On desktop of course it should be possible by simply scrolling.

File: bower.json [hunk 1]
```diff
@@ -28,6 +29,7 @@
     "underscore": "~1.8.3",
     "davclient.js": "https://github.com/evert/davclient.js.git",
     "es6-promise": "^3.2.1",
-    "ical.js": "^1.2.1"
+    "ical.js": "^1.2.1",
+    "web-pull-to-refresh": "git@github.com:ChristophWurst/web-pull-to-refresh.git#^1.1.0"
   }
```
Comment (Gomez) [inline bower.json]: Why is it forked and not used directly?

Comment (ChristophWurst) [inline bower.json]: because the author could remove/rename the repo at any time and that shouldn't break our app

Comment (mar1u5) [inline bower.json]: Would this also happen if we put this in as a GitModule :grin: (Dont know if this would make sense or wether this is possible) ...

Comment (ChristophWurst) [inline bower.json]: put what in a git module?

Comment (mar1u5) [inline bower.json]: `web-pull-to-refresh` :grin:

Comment (ChristophWurst) [inline bower.json]: why should we if we can use bower to manage dependencies? That wouldn't solve any problem but make the install process for devs more complicated

**Comment by ChristophWurst** [comment]
@jancborchardt if we remove the 'Check message' button on top of the messages list, how or where can we should loading feedback? Any idea? I had a quick look at other mail solutions with similar design (Kerio, Outlook online) and to my surprise, they do not have any loading feedback.
On Android the pull to refresh (PTR) of most apps shows a little spinner above (z-axis) the fist element of the list. I doubt that would fit the ownCloud design…

**Comment by jancborchardt** [comment]
> On Android the pull to refresh (PTR) of most apps shows a little spinner above (z-axis) the fist element of the list.

Why would that not fit the design? :) When you pulled down the list, then that's exactly where the feedback should go. Of course the automatic refresh doesn't need feedback in this case.

And yeah, still very important for desktop is that the pull-to-refresh not only works by click-and-drag, but also by simply scrolling up.

**Comment by jancborchardt** [comment]
cc @BernhardPosselt on this one because in the past we experimented with this in the News app and should coordinate there.

**Comment by ChristophWurst** [comment]
JavaScript PTR libs are :hankey:

**Pull request closed without merging**